src/hotspot/os/bsd/os_bsd.cpp

```diff
@@ -1005,27 +1005,44 @@ void *os::Bsd::dlopen_helper(const char *filename, int mode) {
   assert(rtn == 0, "fegetenv must succeed");
 #endif // IA32
 
```

Comment:
Its difficult to see what exactly changed on MacOS. Is this restructuring necessary?

Reply:
I wanted to bring the placemenet of the JFR event and also UL logging of Linux (dlopen_helper)  and BSD/macOS closer together. With the new structure it is more 'uniform' .